### ISSUE TITLE
react/no-multi-comp false positive when wrapping inline stateless with React.memo

### ISSUE THREAD

**Issue by aaronmccall** [yannickcr/eslint-plugin-react#2145]
Given the following code
```javascript
import React from 'react';

const TestStateless = React.memo(props => (
  <div>{props.children}</div>
));

class Test extends React.Component {
  render() {
    return <TestStateless><p>I am wrapped in TestStateless</p></TestStateless>
  }
}
export default Test;
```
and rule config,
```javascript
"react/no-multi-comp": ["error", { "ignoreStateless": true }]
```
I would expect no error, but I get errors as though ignoreStateless were its default of false.

When I define the stateless component outside React.memo then wrap it,
```javascript
import React from 'react';

const stateless = props => <div>{props.children}</div>
const TestStateless = React.memo(stateless);

class Test extends React.Component {
  render() {
    return <TestStateless><p>I am wrapped in TestStateless</p></TestStateless>
  }
}
export default Test;
```
there are no errors.
